Python/optimizer_symbols.c

```diff
@@ -332,13 +431,60 @@ _Py_uop_sym_truthiness(_Py_UopsSymbol *sym)
     return -1;
 }
 
+static JitOptSymbol *
+allocation_base(JitOptContext *ctx)
+{
+    return ctx->t_arena.arena;
+}
+
+JitOptSymbol *
+_Py_uop_sym_new_tuple(JitOptContext *ctx, int size, JitOptSymbol **args)
+{
+    JitOptSymbol *res = sym_new(ctx);
+    if (res == NULL) {
+        return out_of_space(ctx);
+    }
+    if (size > 6) {
+        res->tag = JIT_SYM_KNOWN_CLASS_TAG;
+        res->cls.type = &PyTuple_Type;
+    }
+    else {
+        res->tag = JIT_SYM_TUPLE_TAG;
+        res->tuple.length = size;
+        for (int i = 0; i < size; i++) {
+            res->tuple.items[i] = (uint16_t)(args[i] - allocation_base(ctx));
+        }
+    }
+    return res;
+}
+
+JitOptSymbol *
+_Py_uop_sym_tuple_getitem(JitOptContext *ctx, JitOptSymbol *sym, int item)
+{
+
+    if (sym->tag != JIT_SYM_TUPLE_TAG || item < 0 || item >= sym->tuple.length) {
```

Comment:
We can handle negative indexes pretty easily here, right?

Also, this can probably be updated to work correctly for constant tuples.

Reply:
Negative indexes are errors at this level. If we add a symbolic version of this, we would handle negative indexes there.